packages/router/src/util.ts

```diff
@@ -8,22 +8,40 @@ const createNamedContext = <T>(name: string, defaultValue?: T) => {
 }
 
 /**
- * Get param name and type transform for a route
+ * Get param name, type, and slug for a route.
  *
- *  '/blog/{year}/{month}/{day:Int}/{filePath...}' => [['year'], ['month'], ['day', 'Int'], ['filePath', Glob]]
- *
- * The Glob type is unique, and is used for route params that are strings that
- * include '/' characters, such as a file path. The ternary below handles them
- * differently since the param type is not delimited by a ':'
+ *  '/blog/{year}/{month}/{day:Int}/{filePath...}'
+ *   => [
+ *        ['year',     'String', '{year}'],
+ *        ['month',    'String', '{month}'],
+ *        ['day',      'Int',    '{day:Int}'],
+ *        ['filePath', 'Glob',   '{filePath...}']
+ *      ]
  */
-const paramsForRoute = (route: string) => {
+export const paramsForRoute = (route: string) => {
   // Match the strings between `{` and `}`.
   const params = [...route.matchAll(/\{([^}]+)\}/g)]
 
   return params
     .map((match) => match[1])
     .map((match) => {
-      return match.slice(-3) === '...' ? [match, 'Glob'] : match.split(':')
```

Comment:
Changing this to return `[match.slice(0, -3), 'Glob']` for the glob case would have been enough to fix the behavior, right?

Reply:
Well, almost. There was another place that it had to be tweaked to do the replacement properly, and at that point I wanted to simplify the code, so I did!